src/test/java/org/javarosa/form/api/FormEntryModelTest.java

```diff
@@ -74,4 +74,51 @@ public void isIndexRelevant_respectsRelevanceOfOutermostGroup() throws IOExcepti
         scenario.answer("/data/outerYesNo", "yes");
         assertThat(formEntryModel.isIndexRelevant(q1Index), is(true));
     }
+
+    /**
+     * Identical expressions in a form get collapsed to a single Triggerable and the Triggerable's context becomes
+     * its targets' highest common parent (see Triggerable.intersectContextWith). This makes evaluation in the context
+     * of repeats hard to reason about. This test shows that relevance is propagated as expected when a relevance expression
+     * is shared between a repeat and non-repeat.
+     */
+    @Test
+    public void whenRepeatAndNonRepeatShareRelevanceExpr_repeatContentRelevanceRespectsHierarchy() throws IOException {
```

Comment:
Not sure why we need this and the other test?

Reply:
You mean the other simpler one could be removed and both the general repeat prompt case and the collapsed triggers led case could be covered by this one?

Reply:
Chatted about this and agreed to delete this test.

Reply:
After looking at it some more, I remembered that this test here was an additional one I wrote to verify that the relevance was correctly updated at every level of a hierarchy. It's a bit more coupled with the implementation since it uses `isRepeatRelevant`.

I'm going to add some other tests around Triggerables with multiple targets of different contexts. When I do that, I'll figure out whether this test should go away or stay and if it stays, where it should go.